dev/core#5974 SearchKit - Remove unused dataType from search display columns

colemanw opened the pull request:
Overview
----------------------------------------
Removes unused cruft from SearchDisplays.
See [dev/core#5974](https://lab.civicrm.org/dev/core/-/issues/5974)

Before
----------------------------------------
Unused bit of info stored in every search display.

After
----------------------------------------
Gone.

Technical Details
----------------------------------------
In the early days of SearchKit, rendering was done client-side and required this metadata. Nowadays rendering is all done server-side and the metadata is taken from the schema. So this bit of the search display definition is used.

ufundo commented:
Wait! ChartKit uses this to auto-detect sensible column settings atm :grimacing:

ufundo commented:
I've noticed it's not always exactly what you want (e.g. with option group columns) -- so maybe ChartKit could lookup the metadata separately somehow?

colemanw commented:
@ufundo right, it's not always accurate when added via this js code. So I suggest we go ahead and merge this and then you just need to implement `hook_civicrm_pre` in chartKit to add it via the more accurate php method instead, which can if you wish, put it in the exact same place as before. Here's how SKEntity display does it (putting it in a slightly different place, `$column['spec']['dataType']` but you can simply put it in `$column['dataType`]` to avoid BC problems):

https://github.com/civicrm/civicrm-core/blob/effc650713e6795cfd10d2846419dfe4078431c7/ext/search_kit/Civi/Api4/Event/Subscriber/SKEntitySubscriber.php#L112-L117

mattwire commented:
@ufundo I guess this probably impacts searchkitmaps as well? Since I copied the data type logic from chart_kit

colemanw commented:
Let's implement that hook for the other 2 extensions to make sure they are covered before we merge this.

ufundo commented:
Interesting thanks @colemanw 

If I understand it correctly, this opens the door to getting a more specific bit of metadata for each column in  chart_kit context.

@mattwire yes I think it would affect the maps too, hopefully the same fix should work for both

eileenmcnaughton commented:
Well THIS is stale - @mattwire @colemanw any idea whether we should resuscitate?

colemanw committed:
dev/core#5974 - Remove unused dataType from managed search displays

This metadata is no longer added by SearchKit
so can be removed from managed search displays.

colemanw committed:
dev/core#5974 - Stop adding dataType to search display columns

In the early days of SearchKit, rendering was done client-side and required this metadata.
Nowadays rendering is all done server-side and the metadata is taken from the schema.
So this bit of the search display definition is used.
